File: go/scripts/http_client.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"time"
)

func main() {
	client := &http.Client{
		Timeout: 10 * time.Second,
	}

	url := "https://api.github.com/users/github"

	req, err := http.NewRequest("GET", url, nil)
	if err != nil {
		fmt.Printf("Error creating request: %v\n", err)
		return
	}

	req.Header.Set("User-Agent", "Go HTTP Client")

	resp, err := client.Do(req)
	if err != nil {
		fmt.Printf("Error making request: %v\n", err)
		return
	}
	defer resp.Body.Close()

	fmt.Printf("Status Code: %d\n", resp.StatusCode)

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Printf("Error reading body: %v\n", err)
		return
	}

	var result map[string]interface{}
	if err := json.Unmarshal(body, &result); err != nil {
		fmt.Printf("Error unmarshaling: %v\n", err)
		return
	}

	fmt.Printf("\nUser: %v\n", result["name"])
	fmt.Printf("Bio: %v\n", result["bio"])
	fmt.Printf("Public repos: %v\n", result["public_repos"])
}

File: rust/scripts/http_client.rs
// Note: Add to Cargo.toml:
// [dependencies]
// reqwest = { version = "0.11", features = ["blocking", "json"] }
// serde_json = "1.0"

use reqwest::blocking::Client;
use serde_json::Value;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::new();
    let url = "https://api.github.com/users/github";

    let response = client
        .get(url)
        .header("User-Agent", "Rust HTTP Client")
        .send()?;

    println!("Status Code: {}", response.status());

    let json: Value = response.json()?;

    println!("\nUser: {}", json["name"]);
    println!("Bio: {}", json["bio"]);
    println!("Public repos: {}", json["public_repos"]);

    Ok(())
}
